toucan_connectors/toucan_connector.py

```diff
@@ -26,7 +26,20 @@ class Category(str, Enum):
 
 class DataSlice(NamedTuple):
     df: pd.DataFrame  # the dataframe of the slice
-    total_count: int  # the length of the raw dataframe (without slicing)
+    total_count: Optional[
+        int
+    ] = None  # the length of the raw dataframe (without slicing) TODO remove this one ?
+    total_rows: Optional[int] = None  # total number of rows in original dataset
+    total_returned_rows: Optional[int] = None  # the number of rows returned by the query
+    input_parameters_limit: Optional[
+        int
+    ] = None  # the value of a potential limit extracted from query
+    input_parameters_offset: Optional[
+        int
+    ] = None  # the value of a potential offset extracted from query
+    execution_time: Optional[float] = None  # query's execution time in ms
+    conversion_time: Optional[float] = None  # Result conversion to DataFrame time
+    size: Optional[int] = None  # Dataframe's memory usage in bytes
```

Comment:
can we have something more explicit ? `df_size` ? `df_mem_size`

Reply:
:+1: